### PR TITLE
add macro for modular_adaptor test

### PR DESCRIPTION
**Pull request by AndreyMlashkin**

### CODE REVIEW

**Comment by AndreyMlashkin** [comment]
closes https://github.com/NilFoundation/crypto3-multiprecision/pull/62

File: .github/workflows/pull-request.yml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    ||
+    format('{0}/{1}', github.workflow, github.ref) }}
+  cancel-in-progress: true
+
+jobs:
+  handle-syncwith:
+    name: Call Reusable SyncWith Handler
+    uses: NilFoundation/ci-cd/.github/workflows/reusable-handle-syncwith.yml@v1.2.0
+    with:
+      ci-cd-ref: 'v1.2.0'
```
Comment (x-mass) [inline .github/workflows/pull-request.yml]: v1.2.1

File: test/CMakeLists.txt [hunk 2]
```diff
@@ -467,12 +456,36 @@ macro(define_runtime_multiprecision_test name)
...
+    set(test_name "${CURRENT_PROJECT_NAME}_${name}_test")
+    target_compile_definitions(${test_name} PUBLIC -DTEST_CPP_INT)
+    target_link_libraries(${test_name} no_eh_support)
+    add_dependencies(${CURRENT_PROJECT_NAME}_test_suite_modular_cpp_int_tests ${test_name})
+    set_target_properties(${test_name} PROPERTIES CXX_STANDARD 17)
+    if (CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
+        target_compile_options(${test_name} PRIVATE "-fconstexpr-steps=2147483647")
+    elseif (CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+        target_compile_options(${test_name} PRIVATE "-fconstexpr-ops-limit=4294967295")
+    endif()
```
Comment (x-mass) [inline test/CMakeLists.txt]: This is already set in `define_runtime_multiprecision_test`

File: .github/workflows/pull-request.yml
Comment (x-mass) [inline .github/workflows/pull-request.yml:1]: nit: You moved the set_version.yml, but removed/created all other workflows. Better to keep smaller diff

File: test/modular_adaptor_fixed_test.cpp
Comment (x-mass) [inline test/modular_adaptor_fixed_test.cpp:1]: Aside: I see, that naming as some_name_test.cpp is enforced by CMake `define_runtime_multiprecision_test` func, let's remove `_test` postfix someday though. It is obvious there are tests inside test dir

**Review by x-mass — APPROVED** [review]

File: test/modular_adaptor_fixed_test.cpp
Comment (AndreyMlashkin) [reply test/modular_adaptor_fixed_test.cpp:1]: removed _test suffix

File: .github/workflows/pull-request.yml
Comment (AndreyMlashkin) [reply .github/workflows/pull-request.yml:1]: is it a big problem? I don't see anything repo-specific in set_version, I guess it is safe to add

File: test/CMakeLists.txt (hunk 2 above)
Comment (AndreyMlashkin) [reply test/CMakeLists.txt]: good point, thank you

File: .github/workflows/pull-request.yml (hunk 1 above)
Comment (AndreyMlashkin) [reply .github/workflows/pull-request.yml]: updated

**Pull request merged**